### PR TITLE
Update versions and switch build system

### PR DESCRIPTION
**Pull request by cjbattey**
This PR updates python, all dependency versions, the build system, and installation instructions to address https://github.com/kr-colab/popvae/issues/22. This setup worked for a fresh install on an M3 MacBook, using mamba for the virtual environment. There are no changes to analytical code. 

Sorry about the size of the PR -- it's mostly build stuff and `black` reformatting the code. The relevant changes for installation are switching `setup.py` to `pyproject.toml`, README updates, and some minor refactors of `plotvae.py` to work with a newer version of `bokeh`. 

Changes: 
- switch from setuptools to hatchling for package builds
- switch from setup.py to pyproject.toml
- swith from `python setup.py install` to `pip install ./` for package installation
- upgrade tensorflow and keras to v2.15 and python to 3.10 (surprisingly this involved no changes to the network code, phew)
-  update README with new installation instructions

I tried to get this working with smaller tweaks I think five separate times and kept getting sucked in to more and more confusing dependency issues. But python 3.7 and tensorflow 1 are old enough to be dropping off pip, so have to update. I think (hope) this one will work. 

@andrewkern , any chance you or someone in the lab could give this a shot and see if they can install and run the test data?

### CODE REVIEW

**Comment by andrewkern** [comment]
yo @cjbattey! thanks for doing this. I'll definitely look through it in the next day or so. seems like a lot of code moved here-- mostly auto-reformatting?

**Comment by cjbattey** [comment]
Thanks! Yes, almost all black reformat and build stuff. popvae.py also moved to the top level directory to work with the new build system. But there were no changes in popvae.py code other than import commands to work with a newer Keras version.

File: popvae.py [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+        print("ERROR: `--plot` argument requires `--metadata`")
+        exit()
+
+os.environ["CUDA_VISIBLE_DEVICES"] = gpu_number
+
+if not seed == None:
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    tensorflow.set_random_seed(seed)
```
Comment (andrewkern) [inline popvae.py:179]: this is throwing an error on the example in the readme. I think this call has been changed to `tf.random.set_seed()`

**Review by andrewkern — CHANGES_REQUESTED** [review]
I hit an error running the test example using the `--seed` flag. I think it's a simple enough fix @cjbattey

**Comment by cjbattey** [comment]
oops, sorry about that. Fixed now, and I added a `--test` function. Please pull and give it another shot with `python popvae.py --test` (shouldn't need to reinstall).
